### ISSUE TITLE
toJSON of invalid moment returns "null"

### ISSUE THREAD

**Issue by mj1856** [moment/moment#2886]
We introduced a breaking change in 2.11.0 with #2624, with regard to invalid moments.

```javascript
// 2.10.6
var m = moment("blah","Y");
var s = m.toJSON();  // "Invalid Date"

// 2.11.0
var m = moment("blah","Y");
var s = m.toJSON();  // "null"
```

While we should have been clearer about the breaking change ([apologies](https://twitter.com/wraithgar/status/684085312331452416) to @wraithgar), we also made a mistake in the implementation, and unfortunately it will need to change again.

[src/lib/moment/to-type.js#L31-L34](https://github.com/moment/moment/blob/2.11.0/src/lib/moment/to-type.js#L31-L34) :

```javascript
export function toJSON () {
    // JSON.stringify(new Date(NaN)) === 'null'
    return this.isValid() ? this.toISOString() : 'null';
}
```

This is returning the string `'null'`, not an actual JSON `null` literal.  You can see the difference when used within an object:

```javascript
var o = { d : new Date(NaN), m : moment(NaN) };
var s = JSON.stringify(o);
console.log(s);
```

```json
{"d":null,"m":"null"}
```

Therefore, the `toJSON` function should return `null`, instead of `'null'` for invalid moments.

**Comment by mj1856** [comment]
Consider also that `new Date(NaN).toJSON() === null`.

**Comment by wraithgar** [comment]
Thanks for making this issue, and tracking down the cause while you were at it. It made the PR very easy to write.

**Comment by mj1856** [comment]
Closed in favor of PR #2887, pending next release.  Thanks!
